### ISSUE TITLE
intercept leaks between tests when combined with aliases

### ISSUE THREAD

**Issue by MartijnHols** [cypress-io/cypress#9362]
### Current behavior
Using aliases in combination with intercept causes tests to interfere with other tests regularly.

### Desired behavior
Intercept is cleaned up properly between tests.

### Test code to reproduce
The below test will reproduce this. For me it reproduces it about 60% of the time. I can reproduce it 100% of the time in the closed source test where I encountered the issue, perhaps because it has a lot more assertions. I suppose it may depend on your hardware.

It seems that the error disappears if I clear cookies in `afterEach`.

```ts
const url = 'https://www.cypress.io/'
const dummyRequest = () =>
  cy.request({
    url,
  })

context('InterceptBug', () => {
  beforeEach(() => {
    dummyRequest().as('home')
    cy.visit(url)
  })
  // afterEach(() => {
  //   cy.clearCookies()
  // })

  it('shows the logo', () => {
    cy.get('[alt="Cypress.io"]').should('exist')
  })
  const navigate = () => {
    cy.get('[alt="Cypress.io"]').should('exist')

    // The second time this is called, this sometimes triggers an additional
    // error: `Cannot read property 'fireChangeEvent' of undefined`
    // Other times it will fail on the previous selector with the (assert)
    // error: `expected undefined to exist`
    // I figured those are different bugs, so consider them out of scope for
    // this issue.
    cy.intercept(url).as('navigation')

    cy.contains('Features').click({
      force: true,
    })

    cy.wait('@navigation')

    // If the below visit is removed the test should succeed most of the time
    cy.visit(url)
  }
  it('allows navigating', navigate)
  it('allows navigating2 very irregularly fails', navigate)
  const useHomeAlias = () => {
    cy.get<string>('@home').then((response) => {
      // do something with response
      dummyRequest()
    })
  }
  // cy.get() could not find a registered alias for: @home.
  // You have not aliased anything yet.
  it('use home alias fails', useHomeAlias)
  it('use home alias succeeds', useHomeAlias)
})

export {}
```

### Versions

Cypress version 6.0.0, N/A, Chrome 87, macOS Catalina, N/A

Possibly related issues:

- https://github.com/cypress-io/cypress/issues/8926
- https://github.com/cypress-io/cypress/issues/9170
- https://github.com/cypress-io/cypress/issues/9406

**Comment by jennifer-shehane** [comment]
I can't always get the tests to fail, but it pretty reliably fails with the example below. I could never get the `allows navigating2 very irregularly fails` test to fail, so if you have a simpler example of this, please provide it or open a new issue. 

I'm not totally sure this is related to intercept exactly, it seems like a combination of things are involved. 

### Reproducible Example

The 'use home alias fails' test doesn't seem to be running the `beforeEach` hook.

```js
context('InterceptBug', () => {
  beforeEach(() => {
    cy.request({
      url: 'https://www.cypress.io/',
    }).as('home')
    cy.visit('https://www.cypress.io/')
  })

  it('allows navigating', () => {
    cy.get('[alt="Cypress.io"]')
    cy.intercept('https://www.cypress.io/')
  })

  // cy.get() could not find a registered alias for: @home.
  // You have not aliased anything yet.
  it('use home alias fails', () => {
    cy.get('@home')
  })

  it('use home alias succeeds', () => {
    cy.get('@home')
  })
})
```

<img width="923" alt="Screen Shot 2020-11-30 at 10 07 00 AM" src="https://user-images.githubusercontent.com/1271364/100566280-08718b00-32f4-11eb-8aad-657fc970412e.png">

**Comment by michielmetcake** [comment]
Same issue here. To me it looks like the creation of an alias is skipped if there's already an alias with that name.

Is there a way to flush cache or clear aliasses?

**Comment by jwohlfahrt** [comment]
I'm also seeing this issue intermittently.

**Comment by DoctorBenix** [comment]
Same issue on my project.
Alias value, defined in beforeEach, is recognized in 50% of it() tests, intermittently.
I follow this thread for any update on this issue :)
Thanks

**Comment by mattcat10** [comment]
I am having the same problem reusing aliases

**Comment by jennifer-shehane** [comment]
This has been resolved and is no longer present in our most recent version - 6.2.1

**Comment by jennifer-shehane** [comment]
Just revisiting again. This was fixed in 6.1.0. I'm not exactly sure which PR of that release fixed this.
